wscript

```diff
@@ -198,11 +199,25 @@ def build(bld):
     lib.append('pthread')
 
   clang_tarball_name = None
+  # Fallback for windows
+  default_resource_directory = os.path.join(os.getcwd(), 'resource_dir')
   if bld.env['use_system_clang']:
     rpath = []
+
+    output = subprocess.check_output(['clang', '-###', '-xc', '/dev/null'], stderr=subprocess.STDOUT)
+    match = re.search(r'"-resource-dir" "([^"]*)"', output, re.M | re.I)
+    if match:
+        default_resource_directory = match.group(1)
+    else:
+        print("Failed to found system clang resource directory. Falling back.")
   elif sys.platform.startswith('freebsd') or sys.platform.startswith('linux'):
     clang_tarball_name = os.path.basename(os.path.dirname(bld.env['LIBPATH_clang'][0]))
     rpath = '$ORIGIN/../lib/' + clang_tarball_name + '/lib'
+    default_resource_directory = '../lib/' + clang_tarball_name + '/resource-dir'
+  elif sys.platform == 'darwin':
+    clang_tarball_name = os.path.basename(os.path.dirname(bld.env['LIBPATH_clang'][0]))
+    rpath = '@loader_path/../lib/' + clang_tarball_name + '/lib'
+    default_resource_directory = '../lib/' + clang_tarball_name + '/resource-dir'
```

Comment:
Should `lib/` be `build/`? Why not call the clang in the unzipped folder like above? For me it works on my linux.

Reply:
I suggest a modified version (not sure what's rpath):
```python
  clang_tarball_name = None
  clang_exe = None
  # Fallback for windows
  default_resource_directory = os.path.join(os.getcwd(), 'resource_dir')

  if bld.env['use_system_clang']:
    clang_exe = 'clang'
    rpath = []
  elif sys.platform.startswith('freebsd') or sys.platform.startswith('linux'):
    clang_tarball_name = os.path.basename(os.path.dirname(bld.env['LIBPATH_clang'][0]))
    clang_exe = os.path.join('build', clang_tarball_name, 'bin/clang')
    rpath = os.path.join('$ORIGIN/../lib', clang_tarball_name, 'lib')
  elif sys.platform == 'darwin':
    clang_tarball_name = os.path.basename(os.path.dirname(bld.env['LIBPATH_clang'][0]))
    clang_exe = os.path.join('build', clang_tarball_name, 'bin/clang')
    rpath = os.path.join('@loader_path/../lib', clang_tarball_name, '/lib')
  else:
    rpath = bld.env['LIBPATH_clang']

  if clang_exe:
    output = subprocess.check_output([clang_exe, '-###', '-xc', '/dev/null'],
                                     stderr=subprocess.STDOUT)
    match = re.search(r'"-resource-dir" "([^"]*)"', output, re.M | re.I)
    if match:
      default_resource_directory = match.group(1)
    else:
        print("Failed to found clang resource directory. Falling back.")
```

Reply:
Because, the resource dir is relative to cquery executable, so we have to save the relative path into cquery and let it find the correct path in runtime.

Reply:
`rpath` is the first one where loader find the dynamic libraries upon program start and it is built into the executable. `$ORIGIN` refers to the directory containing the program in linux, and `@loader_path` in macOS.

Reply:
When using system clang, the dynamic loader can always find clang in system directories (e.g. /usr/local/lib), thus we just want to make the resource dir of system clang to be default of the one used in cquery.
When using bundled clang, we want to force cquery to use the bundled clang and bundled resource dir. This is what this PR is all about.

Reply:
I see. Thanks for your explanation!